### PR TITLE
refactor(core): reduce generated code

### PR DESCRIPTION
**Pull request by alan-agius4**
This commit updates the defer code promises to reduce a couple of bytes.

### CODE REVIEW

**Review by thePunderWoman — APPROVED** [review]
LGTM! Thanks!

File: packages/core/src/defer/triggering.ts [hunk 1]
```diff
@@ -444,35 +444,32 @@ function populateHydratingStateForQueue(registry: DehydratedBlockRegistry, queue
...
+  });
 
-    // TODO(incremental-hydration): handle the cleanup for cases when
-    // defer block is no longer present during hydration (e.g. `@if` condition
-    // has changed during hydration/rendering).
-  }
-  return promise;
+  // TODO(incremental-hydration): handle the cleanup for cases when
+  // defer block is no longer present during hydration (e.g. `@if` condition
+  // has changed during hydration/rendering).
```
Comment (AndrewKushnir) [inline packages/core/src/defer/triggering.ts:472]: I think we should move this comment into this block above:
```
  if (deferBlock === null) {
    return;
  }
```

**Review by AndrewKushnir — APPROVED** [review]
@alan-agius4 looks great, just one minor comment. We can also address it in a separate PR.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit c181903c2fe18f1a596643bbe96fcc6eab5c0c60.

The changes were merged into the following branches: main, 19.0.x

**Pull request closed without merging**